lib/metrics.js

```diff
@@ -22,169 +14,65 @@ class MetricsEvent {
-        )
-      )
-    splitStream
-      .pipe(new MergeMetricsStream(...await Promise.all([
-        this.getLatest(),
-        this.getLatest(moduleName)
-      ])))
-      .pipe(new SerializeStream())
-      .pipe(this.logs.createWriteStream(this.allMetricsPath))
-    return deduplicateStream
-  }
-
-  async maybeCreateMetricsFile (moduleName) {
-    await maybeCreateFile(
-      this.#getMetricsFilePath(moduleName),
-      formatLog(
-        JSON.stringify({ totalJobsCompleted: 0, totalEarnings: '0' }) + '\n'
-      )
-    )
-  }
-}
-
-class DeduplicateStream extends Transform {
-  constructor () {
-    super({ objectMode: true })
-    this.last = null
-  }
-
-  _transform (obj, _, callback) {
-    const isChanged = this.last === null ||
-        Object
-          .entries(obj)
-          .some(([key, value]) => this.last[key] !== value)
-    if (isChanged) {
-      this.last = obj
-      callback(null, obj)
-    } else {
-      callback()
-    }
-  }
-}
-
-class TelemetryStream extends Transform {
-  constructor (moduleName) {
-    assert(moduleName, 'moduleName required')
-    super({ objectMode: true })
-    this.lastMetrics = null
-    this.moduleName = moduleName
-  }
-
-  _transform (metrics, _, callback) {
-    if (typeof this.lastMetrics?.totalJobsCompleted === 'number') {
+  submit (moduleName, metrics) {
+    if (
+      typeof this.moduleMetrics.get(moduleName)?.totalJobsCompleted === 'number'
+    ) {
       writeClient.writePoint(
         new Point('jobs-completed')
-          .stringField('module', this.moduleName)
+          .stringField('module', moduleName)
           .intField(
             'value',
-            metrics.totalJobsCompleted - this.lastMetrics.totalJobsCompleted
+            metrics.totalJobsCompleted -
+              this.moduleMetrics.get(moduleName).totalJobsCompleted
           )
       )
     }
-    this.lastMetrics = metrics
-    callback(null, metrics)
-  }
-}
-
-let mergedMetrics
-
-class MergeMetricsStream extends Transform {
-  constructor (allMetrics, moduleMetrics) {
-    assert.strictEqual(
-      typeof allMetrics?.totalJobsCompleted,
-      'number',
-      'allMetrics.totalJobsCompleted required'
-    )
-    assert.strictEqual(
-      typeof moduleMetrics?.totalJobsCompleted,
-      'number',
-      'moduleMetrics.totalJobsCompleted required'
-    )
-    super({ objectMode: true })
-    if (!mergedMetrics) {
-      mergedMetrics = allMetrics
+    this.moduleMetrics.set(moduleName, metrics)
+    const mergedMetrics = {
+      totalJobsCompleted: 0,
+      totalEarnings: '0'
+    }
+    for (const [, metrics] of this.moduleMetrics) {
+      mergedMetrics.totalJobsCompleted += metrics.totalJobsCompleted
+    }
+    const isChanged = this.mergedMetrics === null ||
+      Object
+        .entries(this.mergedMetrics)
+        .some(([key, value]) => mergedMetrics[key] !== value)
+    if (isChanged) {
+      this.mergedMetrics = mergedMetrics
+      this.#events.emit('update', mergedMetrics)
```

Comment:
Can you please add a unit test or two for this code to make sure it really works as intended and does not break in the future?